File: shared/protocol/room_events.dart
/// Canonical SignalR event name constants for the WatchParty room protocol.
/// These mirror RoomEvents.cs exactly. Keep both files in sync.
/// Do not use raw strings — always reference these constants.
library room_events;

class RoomEvents {
  RoomEvents._();

  // Room lifecycle (server → client)
  static const String roomJoined            = 'room:joined';
  static const String roomGuestJoined       = 'room:guest_joined';
  static const String roomGuestLeft         = 'room:guest_left';
  static const String roomGuestReconnected  = 'room:guest_reconnected';
  static const String roomClosed            = 'room:closed';
  static const String roomContentSet        = 'room:content_set';
  static const String roomError             = 'room:error';
  static const String playerReady           = 'player:ready';

  // Playback commands (server → client)
  static const String playbackPlay          = 'playback:play';
  static const String playbackPause         = 'playback:pause';
  static const String playbackSeek          = 'playback:seek';
  static const String playbackStateSync     = 'playback:state_sync';

  // Buffering coordination (server → client)
  static const String bufferingStall        = 'buffering:stall';
  static const String bufferingReady        = 'buffering:ready';
  static const String bufferingResume       = 'buffering:resume';

  // Latency measurement
  static const String pong                  = 'pong';

  // Hub method names (client → server calls)
  static const String hubJoinRoom             = 'JoinRoom';
  static const String hubLeaveRoom            = 'LeaveRoom';
  static const String hubSetContent            = 'SetContent';
  static const String hubPlay                 = 'Play';
  static const String hubPause                = 'Pause';
  static const String hubSeek                 = 'Seek';
  static const String hubNotifyBufferingStall = 'NotifyBufferingStall';
  static const String hubNotifyPlayerReady    = 'NotifyPlayerReady';
  static const String hubNotifyBufferingReady = 'NotifyBufferingReady';
  static const String hubPing                 = 'Ping';
}

File: shared/protocol/RoomEvents.cs
namespace WatchParty.Shared.Protocol;






public static class RoomEvents
{
    
    public const string RoomJoined = "room:joined";
    public const string RoomGuestJoined = "room:guest_joined";
    public const string RoomGuestLeft = "room:guest_left";
    public const string RoomGuestReconnected = "room:guest_reconnected";
    public const string RoomClosed = "room:closed";
    public const string RoomContentSet = "room:content_set";
    public const string RoomError = "room:error";
    public const string PlayerReady = "player:ready";

    
    public const string PlaybackPlay = "playback:play";
    public const string PlaybackPause = "playback:pause";
    public const string PlaybackSeek = "playback:seek";
    public const string PlaybackStateSync = "playback:state_sync";

    
    public const string BufferingStall = "buffering:stall";
    public const string BufferingReady = "buffering:ready";
    public const string BufferingResume = "buffering:resume";

    
    public const string Pong = "pong";

    
    
    public const string HubCreateRoom = "CreateRoom";
    public const string HubJoinRoom = "JoinRoom";
    public const string HubLeaveRoom = "LeaveRoom";
    public const string HubSetContent = "SetContent";
    public const string HubPlay = "Play";
    public const string HubPause = "Pause";
    public const string HubSeek = "Seek";
    public const string HubNotifyBufferingStall = "NotifyBufferingStall";
    public const string HubNotifyBufferingReady = "NotifyBufferingReady";
    public const string HubNotifyPlayerReady = "NotifyPlayerReady";
    public const string HubPing = "Ping";
}
